pkg/bloomgateway/bloomgateway.go

```diff
@@ -334,7 +335,7 @@ func (g *Gateway) FilterChunkRefs(ctx context.Context, req *logproto.FilterChunk
 
 	tasks := make([]Task, 0, len(seriesByDay))
 	for _, seriesWithBounds := range seriesByDay {
-		task, err := NewTask(ctx, tenantID, seriesWithBounds, req.Plan.LineFilterExprs())
+		task, err := NewTask(ctx, tenantID, seriesWithBounds, syntax.ExtractLineFilters(req.Plan.AST))
```

Comment:
If you want to avoid traversing the AST multiple times, you could extract the filters once in the beginning of the request handler.

Reply:
Thank you! Done